File: Frogger/road.cpp
#include "road.h"

Road::Road(float x, float y, float z) : StaticObject(x, y, z) {

	cube = new Cube();
	setupObjects();
}

void Road::setupObjects() {

	float road_amb[] = { 0.05375f, 0.05f, 0.06625f, 1.0f };
	float road_diff[] = { 0.18275f, 0.17f, 0.22525f, 1.0f };
	float road_spec[] = { 0.332741f, 0.328634f, 0.346435f, 1.0f };
	float road_shin[] = { 80.0f };

	cube->addTexture(0, "assets/road3.jpg");
	cube->setColor(VSResourceLib::AMBIENT, road_amb);
	cube->setColor(VSResourceLib::DIFFUSE, road_diff);
	cube->setColor(VSResourceLib::SPECULAR, road_spec);
	cube->setColor(VSResourceLib::SHININESS, road_shin);
}


void Road::render(VSShaderLib shader) {

	vsml->loadIdentity(VSMathLib::MODEL);

	vsml->pushMatrix(VSMathLib::MODEL);
	vsml->translate(position[0], position[1], position[2]);
	vsml->scale(SCALE_FACTOR_X, 1, SCALE_FACTOR_Z);
	cube->render(shader);
	vsml->popMatrix(VSMathLib::MODEL);
}

File: FroggerJS/road.js
function Road(x, y, z) {

	this.x = x;
	this.y = y;
	this.z = z;

	this.cube = new Cube();
	this.setupObjects();
}

Road.prototype.setupObjects = function() {

	this.cube.setColor("ambient", [0.05375, 0.05, 0.06625, 1.0]);
	this.cube.setColor("diffuse", [0.18275, 0.17, 0.22525, 1.0]);
	this.cube.setColor("specular", [0.332741, 0.328634, 0.346435, 1.0]);
	this.cube.setColor("emissive", [0.0, 0.0, 0.0, 1.0]);
	this.cube.setColor("shininess", 80.0);
}

Road.prototype.render = function() {
	
	mat4.identity(mMatrix);
	
	pushMatrix();

	mat4.translate(mMatrix, [this.x, this.y, this.z]);
	mat4.scale(mMatrix, [10.0, 1.0, 100.0]);
	this.cube.render(shaderProgram);

	popMatrix();
}